packages/block-library/src/embed/embed-placeholder.native.js

```diff
@@ -62,9 +60,9 @@ const EmbedPlaceholder = ( {
 			) }
 			<EmbedBottomSheet
 				value={ value }
-				isVisible={ isBottomSheetVisible }
-				onClose={ ( { url } ) => {
-					setIsBottomSheetVisible( false );
+				isVisible={ isEmbedSheeVisible }
+				onClose={ () => setIsEmbedSheetVisible( false ) }
+				onSetAttributes={ ( { url } ) => {
```

Comment:
With the change from using `onClose` to using `onSetAttributes` for setting the attributes, I wonder if you have any thought on which event exactly should cause the preview to reload? I mean, I know the preview is not part of this PR, but I wonder what's the assumption about the API here... would we use the onClose or the URL attribute change as the preview reload event?

Reply:
I haven't checked the preview part yet but I think it should work fine by using the `onSetAttributes` event because, as far as I checked, it's only called when the bottom sheet is closed after changing the URL. However, after reading [your comment about the `onSubmit` callback being called twice](https://github.com/WordPress/gutenberg/pull/31343#discussion_r659792077), I'll review the logic just in case I'm overlooking something. What I'm mostly sure that we shouldn't do is rely on the `onClose` event because after debugging it, I realized that is called multiple times.

Reply:
I posted a comment related to this topic [here](https://github.com/WordPress/gutenberg/pull/31343/files#r659948242), I think the event that we could use for this purpose would be `onDismiss`.

Reply:
Hehe, starting to feel unsure what exactly each of those "similar" props/events actually do and where they differ 🤔. I mean, `onClose` and `onDismiss` sound too close, and, which one would/should call `setAttributes` even? Is it any clearer to you at this Carlos? if so, can you add a line comment about it, and ideally include a relevant explanation in the PR description? Thanks!

Reply:
Yeah agreed, both events sound too close, and to be honest I have the same uncertainty feeling. From what I understand after checking the bottom sheet component, the main differences are:
- `onClose` is triggered when one of the following actions happen (see below), my impression is that this callback is intended for requesting the parent to handle the close flow:
    - Tapping the backdrop element (related to `onBackdropPress` callback).
    - Pressing the back button (only Android) (related to `onBackButtonPress` callback).
    - Perform the escape gesture (related to `onAccessibilityEscape` callback).
    - After the modal is dismissed ([related code](https://github.com/WordPress/gutenberg/blob/83f0f6b0cefe8e1c87a4a2edc118b61f5d5e0620/packages/components/src/mobile/bottom-sheet/index.native.js#L316))
- `onDismiss` on the other hand is triggered when the modal has been already dismissed/hidden, which only happens once in the closing flow.

Probably we should review these flows because I find it weird that `onClose` can be called twice, for example, when tapping the backdrop element (first call) and then when the modal is dismissed (second call).

Reply:
> can you add a line comment about it, and ideally include a relevant explanation in the PR description?

Ideally, we should have a `README` file for documenting the bottom sheet component, I think that would be the best place to explain how the `onClose` and `onDismiss` props work, wdyt? If you agree, I could draft a separate PR for that change.

Reply:
A README would indeed be a better long term place. I guess what we can surface in the meantime, in the context of this PR is the why we went with `onDismiss` for example. That the kind of code design choices that I usually want to have in the PR description to give the needed context for the reader to understand the changes the PR introduces.

Reply:
Right, makes sense 👍 .

I've just updated the PR description adding the insights of the use of both callbacks for this PR in the "Use of onClose and onDismiss callbacks from the bottom sheet component" section. Let me know if the description clarifies this topic and of course any feedback you have, thanks so much for the thorough review 🙇 .